app/code/Magento/Catalog/view/adminhtml/web/js/category-checkbox-tree.js

```diff
@@ -168,7 +168,7 @@ define([
                 }
 
                 if (parent && config && config.length) {
-                    for (i = 0; i < config.length; i++) {
+                    for (var i = 0; i < config.length; i++) {
```

Comment:
Hi @kieronthomas, thank you for collaboration.
According to [Magento Code Style guideline](https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/javascript/js_overview.html)  we should declare a variables in the top of the function.

Reply:
@VladimirZaets nothing to do with me, I think you meant to refer to @keithbentrup ^

Reply:
Ok @VladimirZaets so update it accordingly if you need to. It's a P1 ticket that's crashing browsers. Fix it and move on.

Incidentally, that rule may be part of the reason why the error exists in the first place ... with this perhaps ill-advised rule, declaring vars can be far removed from their use.

For your consideration,
1) https://stackoverflow.com/questions/5053073/is-defining-every-variable-at-the-top-always-the-best-approach
2) https://medium.com/@bluepnume/theres-no-need-to-define-all-javascript-vars-once-at-the-top-of-a-function-and-there-hasn-t-been-a66b31f21822

Also linking to the whole style guide without linking to the specific rule isn't particularly helpful. You're asking contributors to follow rules that you can't even find/link to.

I couldn't even find the rule googling and limiting the domain to just site:devdocs.magento.com. The most relevant section that I could find with a bit of searching which has no mention of declaring it at the top: https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-javascript.html#variable-declarations. You have to go parse the eslint configuration json to actually find the rule you're bringing up.

Lastly, I doubt you're going to fix all the other unrelated errors the picked up by the linter for this file, so arbitrarily fixing it for these and not others seems capricious.  

Anyway, I'm more concerned with this being resolved. You now know the cause of the error, and the solution. You don't need anything further from me to bring it to a resolution.

Reply:
@keithbentrup if there is such rule we cannot violate it. Changing existing conventions is out of scope for PR processing.

@VladimirZaets we can probably move `var` declaration to a proper place on our side as an additional commit.